dynamic_programming/max_sub_array_sum.py

```diff
@@ -0,0 +1,43 @@
+def max_sub_array_sum(arr: list[int], size: int) -> int:
+    """
+    Finds the maximum sum of a subarray within the given array using Kadane's algorithm.
+
+    Args:
+        arr (list): The input array of numbers.
+        size (int): The size of the array.
+
+    Returns:
+        int: The maximum sum of a subarray within the array.
+
+    Example:
+        >>> arr = [-2, -3, 4, -1, -2, 5, -3]
+        >>> max_sub_array_sum(arr, len(arr))
+        6
```

Comment:
Add a blank line to the tests pass.
```suggestion
        6

```
As discussed in `CONTRIBUTING.md` please test your code by running on your local machine:
% `python3 -m doctest -v dynamic_programming/max_sub_array_sum.py`